### ISSUE TITLE
[cmssw_deploy] conficts during application of cherry-picked PRs

### ISSUE THREAD

**Issue by nothingface0** [cms-DQM/fff_dqmtools#29]
Some combinations of CMSSW + PRs seem to prevent clean application of cherry-picked PRs in `cmssw_deploy.py` (introduced in #21).

Examples:

1. ~~`CMSSW_14_1_5_patch1` + https://github.com/cms-sw/cmssw/pull/41932~~ This was using the wrong base release, should use `CMSSW_14_0_2`
3. ~~`CMSSW_14_1_5_patch1` + https://github.com/cms-sw/cmssw/pull/46923~~ This was also dependent on https://github.com/cms-sw/cmssw/pull/46791, applying the latter first, then the former works.

tl;dr it's not an issue of the `cmssw_deploy.py` script
